misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperExecutorUtil.java

```diff
@@ -0,0 +1,424 @@
+    private HashMap<String, Map<String, RecordFieldSymbol>> spreadFieldMap = new HashMap<>();
+    private HashMap<String, String> isOptionalMap = new HashMap<>();
+    private HashMap<String, String> leftFieldMap = new HashMap<>();
+    private HashMap<String, String> responseFieldMap = new HashMap<>();
+    private HashMap<String, String> restFieldMap = new HashMap<>();
+    private HashMap<String, String> spreadFieldResponseMap = new HashMap<>();
+    private ArrayList<String> leftReadOnlyFields = new ArrayList<>();
+    private ArrayList<String> rightSpecificFieldList = new ArrayList<>();
+    private ArrayList<String> optionalRightRecordFields = new ArrayList<>();
+
+    public AIDataMapperExecutorUtil() {
+    }
+
+    public List<TextEdit> generateMappingEdits(ExecuteCommandContext context, JsonArray parameters, Path filePath,
+                                               Range range) throws IOException {
+        List<TextEdit> fEdits = new ArrayList<>();
+
+        fEdits.add(new TextEdit(range, parameters.get(3).getAsString()));
+        JsonArray schemas = (JsonArray) parameters.get(0);
+        String url = parameters.get(1).getAsString();
+        SyntaxTree syntaxTree = context.workspace().syntaxTree(filePath).orElseThrow();
+
+        // Get the last line of the file
+        TextDocument fileContentTextDocument = syntaxTree.textDocument();
+        int numberOfLinesInFile = fileContentTextDocument.toString().split("\n").length;
+        Position startPosOfLastLine = new Position(numberOfLinesInFile + 2, 0);
+        Position endPosOfLastLine = new Position(numberOfLinesInFile + 2, 1);
+        Range newFunctionRange = new Range(startPosOfLastLine, endPosOfLastLine);
+
+        // Get the generated record mapping function
+        String mappingFromServer = getMapping(schemas, url);
+
+        //Read property values
+        JsonElement backgroudObject = parameters.get(2);
+        String generatedRecordMappingFunction = generateMappingFunction(mappingFromServer,
+                (JsonObject) backgroudObject);
+        fEdits.add(new TextEdit(newFunctionRange, generatedRecordMappingFunction));
+        return fEdits;
+    }
+
+    /**
+     * For a give array of schemas, return a mapping function.
+     *
+     * @param schemas {@link JsonArray}
+     * @return mapped function
+     * @throws IOException throws if an error occurred in HTTP request
+     */
+    private String getMapping(JsonArray schemas, String url) throws IOException {
+        int hashCode = schemas.hashCode();
+        if (this.mappingCache.asMap().containsKey(hashCode)) {
+            return this.mappingCache.asMap().get(hashCode);
+        }
+        try {
+            String mappingFromServer = getMappingFromServer(schemas, url);
+            this.mappingCache.put(hashCode, mappingFromServer);
+            return mappingFromServer;
+        } catch (IOException e) {
+            throw new IOException("Error connecting the AI service" + e.getMessage(), e);
```

Comment:
Any reason to append the error message like this since you are passing the error anyway. If it is required then verify the error message ourput.

Reply:
Removed appending the error message